streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java

```diff
@@ -166,30 +167,13 @@ public void openDB(final ProcessorContext context) {
     public void init(final ProcessorContext context,
                      final StateStore root) {
         // open the DB dir
-        this.internalProcessorContext = context;
+        internalProcessorContext = context;
         openDB(context);
-        this.batchingStateRestoreCallback = new RocksDBBatchingRestoreCallback(this);
+        batchingStateRestoreCallback = new RocksDBBatchingRestoreCallback(this);
 
         // value getter should always read directly from rocksDB
         // since it is only for values that are already flushed
-        context.register(root, this.batchingStateRestoreCallback);
-    }
-
-    private RocksDB openDB(final File dir,
-                           final Options options,
-                           final int ttl) throws IOException {
-        try {
-            if (ttl == TTL_NOT_USED) {
```

Comment:
What's up with this TTL thing? It looks like it was always "not used"? (so it's ok to remove it)

Reply:
Yes, it was always "not used". Don't know why we added this code. It's there from the very beginning on. That's why I want to remove it.

Reply:
We do not have TTL support for rocksDB mainly because of its correlations to changelog maintenance. Details here: https://issues.apache.org/jira/browse/KAFKA-4273